### PR TITLE
Fix braces pattern to support more than 2 choices

### LINKED ISSUE #347: `{a,b,c}` no longer considered a dynamic pattern
# Bug Report
Starting with the newly released version `3.2.8` - `{a,b,c}` (sets bigger than 2) is no longer `true` for `isDynamicPattern`, while we only test for `{a,b}` (which still IS considered dynamic).

This broke exactly here:

https://github.com/mrmlnc/fast-glob/pull/344/files#diff-353cf32a130afd9d8b40a67a67f30b51434672caa8ab235726cbd0c82f65fc75R15

### Expected behavior
`isDynamicPattern('{a,b,c}') === true`

### PR DESCRIPTION
**Pull request by amitdahan**
### What is the purpose of this pull request?
Closes #347

### CODE REVIEW

File: src/utils/pattern.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ const COMMON_GLOB_SYMBOLS_RE = /[*?]|^!/;
 const REGEX_CHARACTER_CLASS_SYMBOLS_RE = /\[[^[]*]/;
 const REGEX_GROUP_SYMBOLS_RE = /(?:^|[^!*+?@])\([^(]*\|[^|]*\)/;
 const GLOB_EXTENSION_SYMBOLS_RE = /[!*+?@]\([^(]*\)/;
-const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.)[^,.{]*}/;
+const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.).*[^,.{]*}/;
```
Comment (mrmlnc) [inline src/utils/pattern.ts:15]: According to the CodeQL report, with this change we are returning the polynominal regex. I propose to exclude extra characters from the second part of the expression.

```suggestion
const BRACE_EXPANSIONS_SYMBOLS_RE = /{[^,.{]*(?:,|\.\.)[^{]*}/;
```

File: src/utils/pattern.spec.ts [hunk 2]
```diff
@@ -82,6 +82,8 @@ describe('Utils → Pattern', () => {
 				assert.ok(util.isDynamicPattern('{a,}'));
 				assert.ok(util.isDynamicPattern('{,b}'));
 				assert.ok(util.isDynamicPattern('{a,b}'));
+				assert.ok(util.isDynamicPattern('{a,b,c}'));
+				assert.ok(util.isDynamicPattern('{a' + ','.repeat(999999) + 'b}'));
```
Comment (mrmlnc) [inline src/utils/pattern.spec.ts:86]: I believe we can add tests for nested brackets right now to rule out future breakdowns.

```suggestion
				assert.ok(util.isDynamicPattern('{a' + ','.repeat(999999) + 'b}'));
				assert.ok(util.isDynamicPattern('{a,b,{c,d}}'));
				// The second braces pass
				assert.ok(util.isDynamicPattern('{a,b,{c,d}'));
```

**Review by mrmlnc — CHANGES_REQUESTED** [review]
Thanks for the first contribution 🥇.

I left some suggestions to satisfy CI and eliminate future breakdowns.

File: src/utils/pattern.ts (hunk 1 above)
Comment (amitdahan) [reply src/utils/pattern.ts]: Done! Thanks!

File: src/utils/pattern.spec.ts (hunk 2 above)
Comment (amitdahan) [reply src/utils/pattern.spec.ts:86]: Done!

**Review by mrmlnc — APPROVED** [review]

**Comment by amitdahan** [comment]
Thanks for the super quick review & merge @mrmlnc ! 🚀

**Pull request merged**